File: app/src/main/java/com/example/agiles_3/ReceipeDao.java
package com.example.agiles_3;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ReceipeDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Receipe receipe);

    @Query("SELECT * FROM recetas_tabla ORDER BY calories ASC")
    LiveData<List<Receipe>> getAllReceipes();
}

File: app/src/main/java/com/example/agiles_3/ReceipeDao.kt
package com.example.agiles_3

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface ReceipeDao {
    @Query("SELECT * FROM receta_tabla ORDER BY id ASC")
    fun getSortedId(): Flow<List<Receipe>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(receipe: Receipe)
}